File: Ser2025/Company/Company.Web/Modules/Company/BusinessEntityAddress/BusinessEntityAddressColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.BusinessEntityAddress")]
[BasedOnRow(typeof(BusinessEntityAddressRow), CheckNames = true)]
public class BusinessEntityAddressColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int BusinessEntityId { get; set; }
    public string AddressLine1 { get; set; }
    public string AddressTypeName { get; set; }
    public short IsActive { get; set; }
    public int TenantId { get; set; }
}

File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/BusinessEntityAddressColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { BusinessEntityAddressRow } from './BusinessEntityAddressRow';

export interface BusinessEntityAddressColumns {
    BusinessEntityId: Column<BusinessEntityAddressRow>;
    AddressId: Column<BusinessEntityAddressRow>;
    AddressTypeId: Column<BusinessEntityAddressRow>;
    IsActive: Column<BusinessEntityAddressRow>;
    TenantId: Column<BusinessEntityAddressRow>;
}

export class BusinessEntityAddressColumns extends ColumnsBase<BusinessEntityAddressRow> {
    static readonly columnsKey = 'Company.BusinessEntityAddress';
    static readonly Fields = fieldsProxy<BusinessEntityAddressColumns>();
}